src/libs/runtime-utils.js

```diff
@@ -53,16 +58,22 @@ export const normalizeRuntimeConfig = ({
     // One caveat with using DEFAULT_BOOT_DISK_SIZE here is this over-estimates old GCE runtimes without PD by 1 cent
     // because those runtimes do not have a separate boot disk. But those old GCE runtimes are more than 1 year old if they exist.
     // Hence, we're okay with this caveat.
-    bootDiskSize: bootDiskSize || defaultGceBootDiskSize
+    bootDiskSize: bootDiskSize || defaultGceBootDiskSize,
+    computeRegion: computeRegion || defaultComputeRegion
   }
 }
 
 export const findMachineType = name => {
   return _.find({ name }, machineTypes) || { name, cpu: '?', memory: '?', price: NaN, preemptiblePrice: NaN }
 }
 
-export const getValidGpuTypes = (numCpus, mem) => {
-  const validGpuTypes = _.filter(({ maxNumCpus, maxMem }) => numCpus <= maxNumCpus && mem <= maxMem, gpuTypes)
+export const getValidGpuTypesForZone = zone => {
+  return _.flow(_.find({ name: zone }), _.get(['validTypes']))(zonesToGpus)
+}
+
+export const getValidGpuTypes = (numCpus, mem, zone) => {
+  const validGpuTypesForZone = getValidGpuTypesForZone(zone)
+  const validGpuTypes = _.filter(({ maxNumCpus, maxMem, type }) => numCpus <= maxNumCpus && mem <= maxMem && validGpuTypesForZone.includes(type), gpuTypes)
```

Comment:
I couldn't see where `type` is coming from?

Reply:
Its in the const gpuTypes imported from machines.js, along with maxNumCpus and maxMem.

```
export const gpuTypes = [
  { name: 'NVIDIA Tesla T4', type: 'nvidia-tesla-t4', numGpus: 1, maxNumCpus: 24, maxMem: 156, price: 0.3500, preemptiblePrice: 0.1100 },
...
```